### PR TITLE
Fas 3.0 unittests

### PR DESCRIPTION
**Pull request by skrzepto**
Adding unittests to fas3

It's at about 56% coverage.

How to execute unittests
```
py.test --cov=fas tests/ --cov-report html:cov_html
```

### CODE REVIEW

**Comment by skrzepto** [comment]
Unforunately this is dependent on #175  to get accepted as I branched off of it

File: fas/views/register.py [hunk 1]
```diff
@@ -99,7 +99,8 @@ def account(self):
                     # if peopleform.validate():
                     self.person.avatar = peopleform.avatar.data
                     self.person.introduction = peopleform.introduction.data
-                    self.person.ircnick = peopleform.ircnick.data
+                    self.person.ircnick = peopleform.ircnick.data.strip() \
+                        if peopleform.ircnick.data.strip() else None
```
Comment (pypingou) [inline fas/views/register.py]: This one already has its own PR no?

File: tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+from fas.models.configs import AccountPermissions
+from fas.models.group import (
+    GroupType,
+    Groups,
+    GroupMembership,
+    GroupStatus, MembershipStatus, MembershipRole)
+from fas.models.certificates import PeopleCertificates, Certificates
+from fas.models.la import LicenseAgreement, LicenseAgreementStatus
+
+engine = engine_from_config(settings, 'sqlalchemy.')
```
Comment (pypingou) [inline tests/__init__.py]: Could we see if we can make it run against faitout, maybe on jenkins only then

File: tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    introduction = ['hello', 'intro', None]
+    avatar = [gen_libravatar(email[0]), gen_libravatar(email[1]), None]
+    bio = [None, 'biography of a simple individual', 'i <3 oss']
+    privacy = [1, 0, 1]
+    country_code = [u'US', u'FR', u'CA']
+    latitude = [43, None, Decimal('85.5335165')]
+    longitude = [27, None, Decimal('74.295654')]
+    joined = [datetime(2011, 11, 28, 14, 46, 44),
+              datetime(2013, 8, 28, 14, 46, 44),
+              datetime(2016, 2, 28, 14, 46, 44)]
```
Comment (pypingou) [inline tests/__init__.py]: These likely need to be TZ aware

File: fas/views/register.py (hunk 1 above)
Comment (skrzepto) [inline fas/views/register.py]: Yes it does, I wanted to add those tests in this branch so unfortunately I need to wait for that pr to be accepted. I know this isn't ideal :(

File: tests/__init__.py (hunk 2 above)
Comment (skrzepto) [inline tests/__init__.py]: I think that is possible. I can create a new config called `jenkins.ini` and have it some how loaded. I'll have to think about how to do that. Any thoughts?

File: tests/__init__.py (hunk 3 above)
Comment (skrzepto) [inline tests/__init__.py]: They may, I haven't tested outside my TZ yet so not sure.

File: tests/__init__.py (hunk 2 above)
Comment (pypingou) [inline tests/__init__.py]: On pagure and others, I just check for the environment variable ``BUILD_ID``, cf: https://pagure.io/pagure/blob/master/f/tests/__init__.py#_45-53

File: tests/__init__.py (hunk 3 above)
Comment (pypingou) [inline tests/__init__.py]: Well these datetime are most definitely not TZ aware :)

Comment (pypingou) [inline tests/__init__.py]: A small example:
~~~~
>>> import pytz
>>> import datetime
>>> datetime.datetime.utcnow()
datetime.datetime(2016, 7, 25, 13, 38, 32, 956919)
>>> datetime.datetime.now()
datetime.datetime(2016, 7, 25, 15, 38, 35, 322290)
>>> datetime.datetime(2013, 8, 28, 14, 46, 44)
datetime.datetime(2013, 8, 28, 14, 46, 44)
>>> datetime.datetime.now(pytz.timezone('UTC'))
datetime.datetime(2016, 7, 25, 13, 38, 50, 700129, tzinfo=<UTC>)
~~~~

File: tests/__init__.py (hunk 2 above)
Comment (skrzepto) [inline tests/__init__.py]: Is there a possibility I can set up faitout db now so I can test if it works locally?

Comment (pypingou) [inline tests/__init__.py]: using faitout directly time-out ?

**Comment by skrzepto** [comment]
looks like after rebasing to most recent versionm of FAS_3.0 my unittests are now failing. Need to fix them now

**Comment by skrzepto** [comment]
Hmm it seems that line 295 https://github.com/skrzepto/fas/blob/115507f3748a70313d4c7dbac48f8ae0f1930268/fas/security.py#L295  causes failure. When I commented that line out, more tests pass.

**Comment by skrzepto** [comment]
once pr #196 gets accepted this should pass all the tests

**Comment by skrzepto** [comment]
After #193 gets merged in. Login works fine and I need to update the tests to reflect the changes in that branch.

**Comment by skrzepto** [comment]
Closing this since this needs to be modified for the recent changes in FAS_3 branch

**Pull request closed without merging**